quickstarts/microsoft.datafactory/data-factory-v2-blob-to-blob-copy/main.bicep

```diff
@@ -0,0 +1,142 @@
+@description('Data Factory Name')
+param dataFactoryName string = 'datafactory${uniqueString(resourceGroup().id)}'
+
+@description('Location of the data factory. Currently, only East US, East US 2, and West Europe are supported.')
+param location string = resourceGroup().location
+
+@description('Name of the Azure storage account that contains the input/output data.')
+param storageAccountName string = 'storage${uniqueString(resourceGroup().id)}'
+
+@description('Name of the blob container in the Azure Storage account.')
+param blobContainerName string = 'blob${uniqueString(resourceGroup().id)}'
+
+var dataFactoryLinkedServiceName = 'ArmtemplateStorageLinkedService'
+var DataFactoryDataSetInName = 'ArmtemplateTestDatasetIn'
+var DataFactoryDataSetOutName = 'ArmtemplateTestDatasetOut'
+var pipelineName = 'ArmtemplateSampleCopyPipeline'
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2021-04-01' = {
+  name: storageAccountName
+  location: location
+  sku: {
+    name: 'Standard_LRS'
+  }
+  kind: 'StorageV2'
+}
+
+resource blobContainer 'Microsoft.Storage/storageAccounts/blobServices/containers@2021-04-01' = {
+  name: '${storageAccount.name}/default/${blobContainerName}'
+}
+
+resource dataFactory 'Microsoft.DataFactory/factories@2018-06-01' = {
+  name: dataFactoryName
+  location: location
+  identity: {
+    type: 'SystemAssigned'
+  }
+}
+
+resource dataFactoryLinkedService 'Microsoft.DataFactory/factories/linkedservices@2018-06-01' = {
+  parent: dataFactory
+  name: dataFactoryLinkedServiceName
+  properties: {
+    type: 'AzureBlobStorage'
+    typeProperties: {
+      connectionString: 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};AccountKey=${listKeys(storageAccount.id, storageAccount.apiVersion).keys[0].value}'
+    }
+  }
+}
+
+resource dataFactoryDataSetIn 'Microsoft.DataFactory/factories/datasets@2018-06-01' = {
+  parent: dataFactory
+  name: DataFactoryDataSetInName
+  properties: {
+    linkedServiceName: {
+      referenceName: dataFactoryLinkedServiceName
+      type: 'LinkedServiceReference'
+    }
+    type: 'Binary'
+    typeProperties: {
+      location: {
+        type: 'AzureBlobStorageLocation'
+        container: blobContainer.name
+        folderPath: 'input'
+        fileName: 'emp.txt'
+      }
+    }
+  }
+  dependsOn: [
```

Comment:
Do we need this? Seems like we either don't need it or we need one for the blobContainer too...

Also the resource below has the same pattern and has no dependsOn property

Reply:
used .name notation